Utilities/File.cpp

```diff
@@ -1898,13 +1900,73 @@ bool fs::file::strict_read_check(u64 _size, u64 type_size) const
 	return true;
 }
 
+std::string fs::get_executable_path()
+{
+#if defined(_WIN32)
+	std::vector<wchar_t> buffer(32767);
+	GetModuleFileNameW(nullptr, buffer.data(), buffer.size());
+	return wchar_to_utf8(buffer.data());
+#elif defined(__APPLE__)
+	char bin_path[PATH_MAX];
+	uint32_t bin_path_size = sizeof(bin_path);
+	if (_NSGetExecutablePath(bin_path, &bin_path_size) != 0)
+	{
+		std::cerr << "Failed to find app binary path" << std::endl;
+		return {};
+	}
+
+	// App bundle directory is three levels up from the binary.
+	return get_parent_dir(get_parent_dir(get_parent_dir(bin_path)));
+#else
+	if (const char* appimage_path = ::getenv("APPIMAGE"))
+	{
+		std::cout << "Found AppImage path: " << appimage_path << std::endl;
+		return std::string(appimage_path);
+	}
+
+	std::cout << "Failed to find AppImage path" << std::endl;
```

Comment:
This is how it's supposed to be from the original function. It doesn't know if it's an AppImage or not, but just in case it is, it logs that an AppImage path wasn't found. I can make this message a bit clearer.